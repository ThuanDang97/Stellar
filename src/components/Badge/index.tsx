import React, { memo } from 'react'

// Components
import { BoxShadowStyled } from '@components/styled-components/ViewStyled'
import { BadgeStyled } from '@components/styled-components/TextStyled'

// Props type
import { BadgeProps } from '@self-types/components/Badge.props'

// Utils
import { pxToRem } from '@utils/theme'

// Themes
import { theme } from '@themes/index'

const Badge = ({ title, isReady }: BadgeProps) => (
  <BoxShadowStyled
    display="inline-block"
    shadowType={`${isReady ? 'badgeReadyShow' : 'badgeDefaultShow'}`}
    padding={`${pxToRem(theme.metrics.dimensions.tiny)} ${pxToRem(
      theme.metrics.dimensions.base,
    )}`}
    borderRadius={theme.metrics.borderRadius.base}
    border={`${pxToRem(theme.metrics.borderWidth.default)} solid ${
      isReady ? theme.colors.caribbeanGreen : theme.colors.pizzaz
    }`}
    bgColor={isReady ? theme.colors.foam : theme.colors.bridalHeath}
  >
    <BadgeStyled
      color={isReady ? theme.colors.caribbeanGreen : theme.colors.pizzaz}
    >
      {title}
    </BadgeStyled>
  </BoxShadowStyled>
)

export default memo(Badge)
