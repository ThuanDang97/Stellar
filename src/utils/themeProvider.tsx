import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
