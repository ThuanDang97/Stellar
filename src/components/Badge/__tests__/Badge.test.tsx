import '@testing-library/jest-dom'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

// Component
import Badge from '../index'

describe('Badge Component', () => {
  it('renders Badge component when isReady true', () => {
    const { container } = renderWithTheme(<Badge title="Ready" isReady />)
    expect(container).toMatchSnapshot()
  })

  it('renders Badge component when isReady false', () => {
    const { container } = renderWithTheme(<Badge title="23PTS away" />)
    expect(container).toMatchSnapshot()
  })
})
