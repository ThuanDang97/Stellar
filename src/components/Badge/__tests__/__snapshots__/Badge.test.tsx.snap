// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component renders Badge component when isReady false 1`] = `
<div>
  <div
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 cRwSQg dyHjLC"
    display="inline-block"
  >
    <p
      class="TextStyled__BadgeStyled-sc-9xia41-4 kCrpfs"
      color="#FD8F00"
    >
      23PTS away
    </p>
  </div>
</div>
`;

exports[`Badge Component renders Badge component when isReady true 1`] = `
<div>
  <div
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 bvpxKB fjiEkL"
    display="inline-block"
  >
    <p
      class="TextStyled__BadgeStyled-sc-9xia41-4 ioJMpp"
      color="#00D89D"
    >
      Ready
    </p>
  </div>
</div>
`;
